meson.build

```diff
@@ -334,10 +334,30 @@ if get_option('build_backends')
     'src/neural/cuda/fp16_kernels.cu',
   ]
 
+  required_cuda_header = 'cuda.h'
+  if not cc.has_header(required_cuda_header)
+    cuda_header_found = false
+  endif
+
+  required_cudnn_header = 'cudnn.h'
+  if not cc.has_header(required_cudnn_header)
+    cudnn_header_found = false
+  endif
+
```

Comment:
I think the above block should be split and each half added before the respective `foreach` loop. In most cases `cudnn.h` will be in the same directory with `cuda.h` so it will be found if the check is after the first `foreach` loop.